mainapp/views.py

```diff
@@ -687,10 +686,18 @@ def __init__(self, *args, **kwargs):
         if self.data == {}:
             self.queryset = self.queryset.none()
 
+class VolunteerConsent(UpdateView):
+    model = Volunteer
+    fields = ['has_consented']
+    success_url = '/consent_success/'
+
```

Comment:
Did we miss the actual code? or theres some django magic going on here

Reply:
django magic! it's UpdateView